cmd/tailscale/cli/file.go

```diff
@@ -144,30 +162,78 @@ func runCp(ctx context.Context, args []string) error {
 				return errors.New("directories not supported")
 			}
 			contentLength = fi.Size()
-			fileContents = io.LimitReader(f, contentLength)
+			fileContents = &countingReader{io.LimitReader(f, contentLength), 0}
 			if name == "" {
 				name = filepath.Base(fileArg)
 			}
 
 			if envknob.Bool("TS_DEBUG_SLOW_PUSH") {
-				fileContents = &slowReader{r: fileContents}
+				fileContents = &countingReader{&slowReader{r: fileContents}, 0}
 			}
 		}
 
 		if cpArgs.verbose {
 			log.Printf("sending %q to %v/%v/%v ...", name, target, ip, stableID)
 		}
+
+		var (
+			done = make(chan struct{}, 1)
+			wg   sync.WaitGroup
+		)
+		if isatty.IsTerminal(os.Stderr.Fd()) {
+			go func() {
+				defer wg.Done()
+				var lastBytesRead uint64
+				for {
+					select {
+					case <-done:
+						fmt.Fprintln(os.Stderr)
+						return
+					case <-time.After(time.Second):
+						n := fileContents.ReadBytes()
```

Comment:
this is all indented pretty deep. can you move the goroutine out to its own named func?

Reply:
Done.